libbeat/reader/syslog/syslog.go

```diff
@@ -113,11 +113,8 @@ func ParseMessage(data string, format Format, loc *time.Location) (common.MapStr
 	case FormatRFC5424:
 		m, err = parseRFC5424(data)
 	}
-	if err != nil {
-		return common.MapStr{}, time.Time{}, err
-	}
 
-	return m.fields(), m.timestamp, nil
+	return m.fields(), m.timestamp, err
```

Comment:
Can we differentiate error states that are recoverable from error states that give us no usable information. Also add a not on the godoc to the effect that the error doesn't necessarily preclude use of the fields.

Reply:
At the moment, I don't think so. The parser isn't set up to recover from an error state yet. It can either try again for the exact same result, or it tumbles out with a bunch of errors and partially filled fields, all depends on where and how it fails. That's something I plan to address in the next round of changes.

Reply:
Thanks.